colcon_cmake/task/cmake/__init__.py

```diff
@@ -208,3 +210,55 @@ def get_visual_studio_version():
     :rtype: str
     """
     return os.environ.get('VisualStudioVersion', None)
+
+
+def _parse_cmake_version():
+    """
+    Parse the CMake version number by running `cmake --version`.
+
+    The version number is parse from the output of 'cmake --version' and is
+    expected in the form 'cmake version #.#.#' where each '#' character
+    represents part of the version number. Additional text may follow, however
+    only the first line is parsed. Parsing is performed by the pkg_resources
+    package and the returned object is a Version object from that package.
+
+    This function blocks on the execution of cmake and should only be used to
+    cache the CMake version number. External API users should use
+    get_cmake_version()
+
+    :returns: The version as parsed by the pkg_resources package
+    :rtype pkg_resources.extern.packaging.version.Version
+    """
+    output = subprocess.run([CMAKE_EXECUTABLE, '--version'],
+                            capture_output=True)
+    if output.returncode:
+        # Failed to execute 'cmake --version'
+        return None
+    lines = output.stdout.decode().splitlines()
+    ver_line = lines[0] if lines and len(lines) else None
+    if ver_line:
+        # Extract just the version part of the string.
+        ver_re_str = r'^(?:.*)(\d+\.\d+\..*)'
+        ver_match = re.match(ver_re_str, ver_line, re.I)
+        if ver_match:
+            return parse_version(ver_match.group(1))
+    # Failed to parse version number.
+    return None
+
+
+# Global variable for the cached CMake version number.
+_cached_cmake_version = _parse_cmake_version()
```

Comment:
Currently this determines the CMake version at import time of this module. Please change it to be lazily computed on the first invocation of `get_cmake_version` (as in the previously referenced example).

Reply:
Now lazy.

BTW, should I be clicking `Resolve conversation` or should you?